src/Arc.cpp

```diff
@@ -31,8 +31,16 @@ CArc::~CArc()
 }
 void CArc::Update( cBoundingRectangle& rect )
 {
-    uint8_t start_quadrant = (int)sa % 360 / 90 + 1;// the quadrant of sa
-    uint8_t end_quadrant = (int)ea % 360 / 90 + 1;// the quadrant of ea
+    int8_t start_quadrant = (int)sa % 360 / 90 + 1;// the quadrant of sa
```

Comment:
You are doing rather a lot of casting between different integer types.  Why?  Surely, using int everywhere would make you code easier to understand and slightly faster?

Reply:
yes, that make sense.